roles/add_livemigration_user/tasks/main.yml

```diff
@@ -3,13 +3,38 @@
 
 ---
 - block:
+  - name: Extract distribution name
+    set_fact:
+      base_distribution: >-
+        {%- if ansible_distribution | regex_search("Debian") -%}
+        Debian
+        {%- elif ansible_distribution | regex_search("CentOS") -%}
+        CentOS
+        {%- elif ansible_distribution | regex_search("Yocto") -%}
+        Yocto
+        {%- endif -%}
+  - name: Define specific variables by distribution
+    set_fact:
+      livemigration_config:
+        Debian:
+          groups: "libvirt"
+          create_home: true
+          ssh_path: "/home/{{ livemigration_user }}/.ssh"
+        CentOS:
+          groups: "libvirt"
+          create_home: true
+          ssh_path: "/home/{{ livemigration_user }}/.ssh"
+        Yocto:
+          groups: "qemu,haclient,libvirt"
+          create_home: false
+          ssh_path: "/etc/ssh/{{ livemigration_user }}"
```

Comment:
This is not how we define distribution specific variable in a role.
You should add an `- include_vars: "{{ ansible_distribution | replace(' ', '_') }}.yml"` task at the beginning of the `tasks/main.yml file`, and then create the `CentOS.yml  Debian.yml  Seapath_host_Yocto_distribution.yml` files in the role's `vars` folder

Reply:
Thanks !